### ISSUE TITLE
singularity pull from AWS ECR fails

### ISSUE THREAD

**Issue by tbugfinder** [apptainer/singularity#2059]
## Version of Singularity:

2.6.0-dist

### Expected behavior

`singularity pull` should pull images from AWS ECR successfully.

### Actual behavior

`singularity pull` fails with error message `ERROR Unrecognized authentication challenge, exiting.`.
I've configured SINGULARITY_DOCKER_PASSWORD and SINGULARITY_DOCKER_USERNAME with output of `aws ecr get-login --no-include-email `.
```
# singularity -vvv pull docker://<accountid>.dkr.ecr.<region>.amazonaws.com/<path,image>:tag
Increasing verbosity level (4)
Singularity version: 2.6.0-dist
Exec'ing: /usr/libexec/singularity/cli/pull.exec
Singularity Image: ./<name>.simgWARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
ERROR Unrecognized authentication challenge, exiting.
Cleaning up...
ERROR: pulling container failed!

```

### Steps to reproduce behavior

- set `SINGULARITY_DOCKER_PASSWORD` and `SINGULARITY_DOCKER_USERNAME`
- run `singularity pull docker://< aws ecr >/`

**Comment by tbugfinder** [comment]
verbose output:
```
VERBOSE Registry: <accountid>.dkr.ecr.<region>.amazonaws.com
VERBOSE Namespace: <folder>
VERBOSE Repo Name: <name>
VERBOSE Repo Tag: <repotag>
VERBOSE Version: None
VERBOSE Obtaining tags: https://<accountid>.dkr.ecr.<region>.amazonaws.com/v2/<folder>/<name>/tags/list
DEBUG GET https://<accountid>.dkr.ecr.<region>.amazonaws.com/v2/<folder>/<name>/tags/list
DEBUG Http Error with code 401
DEBUG update_token: challenge = 'Basic realm="https://<accountid>.dkr.ecr.<region>.amazonaws.com/",service="ecr.amazonaws.com"'
ERROR Unrecognized authentication challenge, exiting.
Cleaning up...
ERROR: pulling container failed!
```

**Comment by tbugfinder** [comment]
Also failed using sregistry "client" [https://github.com/singularityhub/sregistry-cli/issues/145](https://github.com/singularityhub/sregistry-cli/issues/145)

**Comment by tbugfinder** [comment]
@bauerm97 Are there any plans to support a private docker registry with authentication?

**Comment by tbugfinder** [comment]
seems to be fixed with ~3.1.x (didn't test 3.0.x though).
